src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -341,7 +353,7 @@ protected boolean isSnapshotRequired(GetCheckpointResponse getCheckpointResponse
         // A call to set the checkpoint is required first otherwise we will get an error 
         // from the server side saying:
         // INTERNAL_ERROR[code 21]: Stream ID {} is expired for Tablet ID {}
-        setCheckpointWithRetryBeforeSnapshot(tableId, tabletId);
+        setCheckpointWithRetry(tableId, tabletId, 0, 0, true, false, 0L);
```

Comment:
1. Let's do one thing, rename this method to `markStreamActive`

2. I think a more preferable way to do that would be to use just the boolean condition, for example, `bootstrap` and remove the redundant parameters which can be inferred from the flag itself:

```java
public void markStreamActive(String tableId, String tabletId, bootstrap)  {
  try {
    long term = 0;
    long index = 0;
    if (bootstrap) {
      term = -1;
      index = -1;
    }
 
    YBClientUtils.setCheckpoint(this.syncClient, this.connectorConfig.streamId(), tableId, tabletId, term,
                                index, true /* initialCheckpoint */, bootstrap);
  } catch (Exception e) {
    // Exception handling
  }
}

```